### PR TITLE
support for Kotlin

### PR DESCRIPTION
**Pull request by rabidaudio**
Based on the discussion on #37 I made some minor changes to support Kotlin coverage.

I think it might need a little work and maybe a test, but I haven't worked on a Gradle plugin before so I'm not really sure how to write a test for it.

Also, it breaks the logging of `classDirectories` because it needs to use two file trees. There may be a way around this but I couldn't figure it out short of printing every file.

### CODE REVIEW

**Comment by SebRut** [comment]
Could this be re-run/manually evaluated as it seems like the CI failed because of an unrelated connection issue.

File: src/main/groovy/com/dicedmelon/gradle/jacoco/android/JacocoAndroidPlugin.groovy [hunk 1]
```diff
@@ -45,6 +45,10 @@ class JacocoAndroidPlugin implements Plugin<ProjectInternal> {
     plugin
   }
 
+  private static boolean hasKotlin(PluginContainer plugins) {
```
Comment (SebRut) [inline src/main/groovy/com/dicedmelon/gradle/jacoco/android/JacocoAndroidPlugin.groovy]: Is that method really necessary for one call?

File: src/main/groovy/com/dicedmelon/gradle/jacoco/android/JacocoAndroidPlugin.groovy [hunk 2]
```diff
@@ -100,7 +113,6 @@ class JacocoAndroidPlugin implements Plugin<ProjectInternal> {
   private void logTaskAdded(JacocoReport reportTask) {
     logger.info("Added $reportTask")
     logger.info("  executionData: $reportTask.executionData.asPath")
-    logger.info("  classDirectories: $reportTask.classDirectories.dir.path")
```
Comment (SebRut) [inline src/main/groovy/com/dicedmelon/gradle/jacoco/android/JacocoAndroidPlugin.groovy]: Why was this removed?

File: src/main/groovy/com/dicedmelon/gradle/jacoco/android/JacocoAndroidPlugin.groovy (hunk 1 above)
Comment (rabidaudio) [reply src/main/groovy/com/dicedmelon/gradle/jacoco/android/JacocoAndroidPlugin.groovy]: It's really not. I was just trying to match the style of [these](https://github.com/fixdauto/jacoco-android-gradle-plugin/blob/ddbd538522f1c2436e522797c4fec85ededf7ec0/src/main/groovy/com/dicedmelon/gradle/jacoco/android/JacocoAndroidPlugin.groovy#L101) (also only used once) but I have no problem in-lining it.

File: src/main/groovy/com/dicedmelon/gradle/jacoco/android/JacocoAndroidPlugin.groovy [hunk 3]
```diff
@@ -64,15 +68,24 @@ class JacocoAndroidPlugin implements Plugin<ProjectInternal> {
...
         JacocoReport)
     reportTask.dependsOn testTask
     reportTask.group = "Reporting"
     reportTask.description = "Generates Jacoco coverage reports for the ${variant.name} variant."
     reportTask.executionData = project.files(executionData)
     reportTask.sourceDirectories = project.files(sourceDirs)
-    reportTask.classDirectories =
+    def javaTree =
         project.fileTree(dir: classesDir, excludes: project.jacocoAndroidUnitTestReport.excludes)
+    if (kotlin) {
```
Comment (angry-programmer) [inline src/main/groovy/com/dicedmelon/gradle/jacoco/android/JacocoAndroidPlugin.groovy]: This if-else statement might cause some problems in mixed Java/Kotlin projects. Don't you think?

Comment (rabidaudio) [reply src/main/groovy/com/dicedmelon/gradle/jacoco/android/JacocoAndroidPlugin.groovy]: Would it? It uses both java and kotlin trees. Unless the same classes up in both places?

Comment (rabidaudio) [reply src/main/groovy/com/dicedmelon/gradle/jacoco/android/JacocoAndroidPlugin.groovy]: Our project is mixed and I haven't noticed any issues

**Comment by peterfortuin** [comment]
Are there plans to finish this pull request?

**Comment by rabidaudio** [comment]
@peterfortuin If you have zero coverage, it sounds like a configuration problem. We still have some problems with running within Android Studio but running the gradle task works with the correct coverage.

This is our configuration, if it is helpful:

```groovy
apply plugin: 'jacoco-android'

android {
    testOptions.unitTests.all {
        jacoco { includeNoLocationClasses = true }
    }
}

jacoco {
    toolVersion = "0.8.2"
}

jacocoAndroidUnitTestReport {
    csv.enabled false
    html.enabled true
    xml.enabled true

    excludes += ['jdk.internal.*']
}
```

**Comment by arturdm** [comment]
Fixes #37 
Fixes #26 

Released as part of [0.1.4](https://github.com/arturdm/jacoco-android-gradle-plugin/releases/tag/jacoco-android-release-0.1.4). Thank you for contributing!
